zcash_client_sqlite/src/wallet.rs

```diff
@@ -380,16 +381,17 @@ pub(crate) fn get_transparent_receivers<P: consensus::Parameters>(
             })?;
 
         if let Some(taddr) = ua.transparent() {
-            ret.insert(
-                *taddr,
-                AddressMetadata::new(account, DiversifierIndex::from(di_be)),
-            );
+            let di_short = DiversifierIndex::from(di).try_into();
```

Comment:
```suggestion
            let di_short: Result<u32, _> = DiversifierIndex::from(di).try_into();

Reply:
If I read this correctly, it's just adding a type annotation that matches what rust was already inferring, is that right? If so, what's your general guidance for when to add optional type annotations?

Reply:
This code was rewritten anyway.

Reply:
Yes the suggestion was for what Rust was already inferring. To answer your question, I typically add type annotations when:
* inferring the type would require a non-local search (in this case, for implementors of `From<DiversifierIndex>`); *and*
* the type is useful to know in order to understand the code.

Uses of `into()`, `try_into()`, and `collect()` are particularly common cases (but not all of them).